### PR TITLE
First pass of basic FE test coverage

### PR DESCRIPTION
**Pull request by jlhogan**
adds snapshot and axe accessibility checks to all containers

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -3,10 +3,11 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@ctoec/component-library": "https://github.com/ctoec/component-library#base",
+    "@ctoec/component-library": "https://github.com/ctoec/component-library#aria-required",
```
Comment (jlhogan) [inline client/package.json:6]: TODO: Update to `base` when feature branch lands

File: client/package.json [hunk 2]
```diff
@@ -42,7 +43,7 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
+    "test": "CI=true react-scripts test --env=jest-environment-jsdom-sixteen",
```
Comment (jlhogan) [inline client/package.json:46]: necessary to get some of the `dom` utils to work

File: client/package.json [hunk 3]
```diff
@@ -59,5 +60,12 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
```
Comment (jlhogan) [inline client/package.json:64]: should these just be normal deps?

File: client/package.json [hunk 4]
```diff
@@ -59,5 +60,12 @@
...
+    "@types/jest-axe": "^3.5.0",
+    "jest-axe": "^4.0.0",
+    "jest-canvas-mock": "^2.2.0",
```
Comment (jlhogan) [inline client/package.json:67]: needed to get rid of a warning from some dep

File: client/src/containers/CheckData/CheckData.tsx [hunk 5]
```diff
@@ -17,7 +17,7 @@ const CheckData: React.FC = () => {
   const [reportData, setReportData] = useState<Child[]>([]);
 
   useEffect(() => {
-    if (reportId && accessToken) {
```
Comment (jlhogan) [inline client/src/containers/CheckData/CheckData.tsx:20]: the `api` util does this check; no need to do it 2x

File: client/src/containers/App/App.test.tsx [hunk 6]
```diff
@@ -2,13 +2,15 @@ import React from 'react';
...
-  );
-  expect(asFragment()).toMatchSnapshot();
+const routerWrapped = (
+  <BrowserRouter>
+    <App />
+  </BrowserRouter>
+);
+
+describe('App', () => {
+  snapshotTestHelper(routerWrapped);
```
Comment (mmazanec22) [inline client/src/containers/App/App.test.tsx:14]: Nice!

File: client/src/containers/CheckData/CheckData.test.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import {
+  waitFor,
+  waitForElementToBeRemoved,
+  render,
+} from '@testing-library/react';
+import { Child, FundingSource } from '../../shared/models';
+import CheckData from './CheckData';
+import { accessibilityTestHelper, snapshotTestHelper } from '../../testHelpers';
+
+const routerWrapped = (
```
Comment (mmazanec22) [inline client/src/containers/CheckData/CheckData.test.tsx:12]: Should we add a "wrap in router" test helper too maybe?

File: client/src/containers/CheckData/CheckData.tsx (hunk 5 above)
Comment (mmazanec22) [reply client/src/containers/CheckData/CheckData.tsx:20]: Would keeping access token here help prevent unnecessary network requests/is that worth trying to do?

**Review by mmazanec22 — APPROVED** [review]

File: client/src/containers/CheckData/CheckData.tsx (hunk 5 above)
Comment (jlhogan) [reply client/src/containers/CheckData/CheckData.tsx:20]: api util does the check -- if accessToken is provided, but has no value (aka is null as opposed to undefined) then the api helper doesn't actually make the network request https://github.com/ctoec/data-collection/blob/base/client/src/utils/api.ts#L67

**Pull request merged**